modules/shared/programs/claude/files/skills/analyzing-da-sessions/scripts/analyze.py

```
@@ -0,0 +1,915 @@
+def collect_remote_files(host: str, warnings: list[str]) -> list[str]:
+    """원격 호스트에서 jsonl 파일 path glob (subprocess.run 고정 argv).
+
+    원격 find stdout의 path 라인은 비신뢰 입력으로 간주하여, _allowed_remote_path가 통과한
+    line만 수집한다 — 제어문자/shell metacharacter 포함 line은 silently 폐기한다.
+    """
+    _validate_host(host)
+    paths = HOST_PATH_MAP[host]
+    all_files: list[str] = []
+    for base in (paths["claude"], paths["codex"]):
+        try:
+            proc = subprocess.run(
```

Ruff (0.15.12)

[error] `subprocess` call: check for execution of untrusted input

(S603)